clang/include/clang/CIR/Interfaces/ASTAttrInterfaces.td

```diff
@@ -0,0 +1,213 @@
+//===- ASTAttrInterfaces.td - CIR AST Interface Definitions -----*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_CIR_INTERFACES_AST_ATTR_INTERFACES
+#define MLIR_CIR_INTERFACES_AST_ATTR_INTERFACES
+
+include "mlir/IR/OpBase.td"
+
+let cppNamespace = "::mlir::cir" in {
+
+    def ASTDeclInterface : AttrInterface< "ASTDeclInterface" > {
+        let methods = [
+            InterfaceMethod< "", "bool", "hasOwnerAttr", (ins), [{}],
+                /*defaultImplementation=*/ [{
+                    return $_attr.getAstDecl()->template hasAttr< clang::OwnerAttr >();
+                }]
+            >,
+            InterfaceMethod< "", "bool", "hasPointerAttr", (ins), [{}],
+                /*defaultImplementation=*/ [{
+                    return $_attr.getAstDecl()->template hasAttr< clang::PointerAttr >();
+                }]
+            >,
+            InterfaceMethod< "", "bool", "hasInitPriorityAttr", (ins), [{}],
+                /*defaultImplementation=*/ [{
+                    return $_attr.getAstDecl()->template hasAttr< clang::InitPriorityAttr >();
+                }]
+            >
+        ];
+    }
+
+    def ASTDeclContextInterface : AttrInterface< "ASTDeclContextInterface" > {
+        let methods = [
+            InterfaceMethod< "",
+                "llvm::SmallVector< ::mlir::Attribute, 4 >", "decls",
+                (ins "::mlir::MLIRContext *":$ctx), [{}],
+                /*defaultImplementation=*/ [{
+                    llvm::SmallVector< mlir::Attribute, 4 > decls;
+                    for (const auto *sub : $_attr.getAstDecl()->decls()) {
+                        decls.push_back(makeAstDeclAttr(sub, ctx));
+                    }
+                    return decls;
+                }]
+            >
+        ];
+    }
+
+    def ASTNamedDeclInterface : AttrInterface< "ASTNamedDeclInterface",
+        [ASTDeclInterface]
+    > {
```

Comment:
More style comments: `> {` should be above, not after the newline.

Reply:
fixed in a6c1d8c990aa849c0a88db914e1af2c858259ae0